src/Ushahidi/Modules/V5/Http/Controllers/PermissionsController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Http\Controllers;
+
+use App\Bus\Query\QueryBus;
+use Ushahidi\Modules\V5\Http\Resources\Permissions\PermissionsCollection;
+use Ushahidi\Modules\V5\Http\Resources\Permissions\PermissionsResource;
+use Illuminate\Http\Request;
+use Ushahidi\Modules\V5\Models\Permissions;
+use Ushahidi\Modules\V5\Actions\Permissions\Queries\FetchPermissionsByIdQuery;
+use Ushahidi\Modules\V5\Actions\Permissions\Queries\FetchPermissionsQuery;
+
+class PermissionsController extends V5Controller
+{
+
+      
+    /**
+     * Display the specified resource.
+     *
+     * @param integer $id
+     * @return mixed
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function show(int $id, QueryBus $queryBus)
+    {
+        $permissions = $queryBus->handle(new FetchPermissionsByIdQuery($id));
+       // $this->authorizeForCurrentUser('show', $permissions);
+        return new PermissionsResource($permissions);
+    }//end show()
+
+   
+    /**
+     * Display the specified resource.
+     *
+     * @return TosCollection
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function index(Request $request, QueryBus $queryBus)
+    {
+       // $this->authorizeForCurrentUser('index', Permissions::class);
+        $resourceCollection = new PermissionsCollection(
+            $queryBus->handle(
+                new FetchPermissionsQuery(
```

Comment:
I think we could use the FetchPermissionsQuery constants as a default values in the constructor itself, here: https://github.com/ushahidi/platform/pull/4509/files#diff-12be26afa959905d2b900fbfc37c45b829e65e705d4b8ecffa5c0256618e4907R20

`$request->query('foo')` returns null if it won't find the value inside the query string so constructor would default to a specific value set in the method declaration, and that way we won't have to think about it next time we use `FetchPermissionsQuery`

Reply:
I tried to do as you suggest but when it constructor doesn't take the default value when passing null.